File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/observer/solution/Manager.m
% Manager.m

classdef Manager < Participant
	methods
		function add(theManager, aMeeting)
			theManager.add@Participant(aMeeting);

			fprintf('Manager.add(%s)\n', aMeeting.toString);
		end

		function remove(theManager, aMeeting)
			theManager.remove@Participant(aMeeting);

			fprintf('Manager.remove(%s)\n', aMeeting.toString);
		end

		function update(theManager, aMeeting)
			theManager.update@Participant(aMeeting);

			fprintf('Manager.update(%s)\n', aMeeting.toString);
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/observer/solution/Manager.java
// Manager.java

public class Manager extends Participant
{
	public void add(Meeting aMeeting)
	{
		super.add(aMeeting);

		System.out.println(
			  "Manager.add("
			+ aMeeting.toString()
			+ ")"
			);
	}

	public void remove(Meeting aMeeting)
	{
		super.remove(aMeeting);

		System.out.println(
			  "Manager.remove("
			+ aMeeting.toString()
			+ ")"
			);
	}

	public void update(Meeting aMeeting)
	{
		super.update(aMeeting);

		System.out.println(
			  "Manager.update("
			+ aMeeting.toString()
			+ ")"
			);
	}
}
